Training5/manageapp/src/redux/saga/userSaga.js

```diff
@@ -0,0 +1,55 @@
+import { put, call, takeEvery } from "redux-saga/effects";
+import axios from "axios";
+import { ACTION_NAMES } from "../action/ActionNames";
+import { BASE_URL } from "../../utils/Config";
+function* LoginUser(action) {
+    try {
+        const result = yield call(axios.post, `${BASE_URL}login`, {
+            email: action.data.email,
+            password: action.data.password,
+        });
+        const data = result.data;
+        if (data.error !== "Incorrect password or email") {
+            yield put({ type: "SAGA_LOGIN_USER", data: data });
+            window.location.replace("/home");
+        }
+    } catch (err) {
+        console.error(err);
+    }
+}
+export function* LoginUserWatch() {
+    yield takeEvery(ACTION_NAMES.LOGIN_USER, LoginUser);
+}
+function* getAllUsersInfo(action) {
+    try {
+        const result = yield call(axios.get, `${BASE_URL}api/users`, {
```

Comment:
@Huynhkimphat  using axios interceptor & store to attach token into the header in every API call to our backend, should not manually on each saga function